pkg/customresources/keymanagementproviders/kmpcertmanager.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package keymanagementproviders
+
+import (
+	"crypto/x509"
+	"sync"
+
+	"github.com/deislabs/ratify/internal/constants"
+	kmp "github.com/deislabs/ratify/pkg/keymanagementprovider"
+)
+
+// ActiveCertStores implements the KMPCertManager interface.
+type ActiveCertStores struct {
```

Comment:
Thanks for the work here @binbin-li! I'm confused on why we need this extra layer of namespaced mapping? The existing implementation already has support for namespace built in though?

Reply:
That's good point. I was also considering if I can utilize the existing mapping for namespace use case. My concern is 
1. certificatesMap and keyMap have key in format: `namespace/name`, I'm thinking if we would need operations on `namespace` only in the future.
2. Other resources have similar mapping layout, namespace->name->resource, so I feel it would be easier to understand.

But I do admit that there is redundancy between mapping `namespace` to `namespace/name` and complicates the operations as we have nested sync.Map.

Since you're working on Cosign enhancement, if we don't have plan for `namespace` key usage, I can remove the extra namespace mapping.